src/librbd/migration/NativeFormat.cc

```diff
@@ -61,11 +64,71 @@ template <typename I>
 void NativeFormat<I>::open(Context* on_finish) {
   ldout(m_cct, 10) << dendl;
 
+  std::optional<std::string> cluster_name;
+  auto& cluster_name_val = m_json_object[CLUSTER_NAME_KEY];
+  if (cluster_name_val.type() == json_spirit::str_type) {
+    cluster_name = cluster_name_val.get_str();
+  } else if (cluster_name_val.type() != json_spirit::null_type) {
+    lderr(m_cct) << "invalid cluster name" << dendl;
+    on_finish->complete(-EINVAL);
+    return;
+  }
+
+  std::optional<std::string> client_name;
+  auto& client_name_val = m_json_object[CLIENT_NAME_KEY];
+  if (client_name_val.type() == json_spirit::str_type) {
+    if (!cluster_name.has_value()) {
+      lderr(m_cct) << "client_name cannot be used without cluster_name"
+                   << dendl;
+      on_finish->complete(-EINVAL);
+      return;
+    }
+    client_name = client_name_val.get_str();
+  } else if (client_name_val.type() != json_spirit::null_type) {
+    lderr(m_cct) << "invalid client name" << dendl;
+    on_finish->complete(-EINVAL);
+    return;
+  } else {
+    if (cluster_name.has_value()) {
+      lderr(m_cct) << "client_name must be specified alongside cluster_name"
+                   << dendl;
+      on_finish->complete(-EINVAL);
+      return;
+    }
+  }
+
+  librados::Rados rados(m_image_ctx->md_ctx);
+  librados::Rados* rados_ptr = &rados;
+  if (cluster_name.has_value()) {
+    std::vector<const char*> args;
+    int r = util::init_rados(m_cct, cluster_name.value(), client_name.value(),
+                             "", "", "external cluster", args,
+                             &rados_ptr, true, "");
+    if (r < 0) {
+      lderr(m_cct) << "cannot initialize external cluster client:"
+                   << cpp_strerror(r) << dendl;
+      on_finish->complete(r);
+      return;
+    }
+
+    // disable unnecessary librbd cache
+    auto cct = (CephContext*)(rados_ptr->cct());
+    r = cct->_conf.set_val("rbd_cache", "false");
```

Comment:
Actually I am not sure we need to do this here. I mean I don't know how useful it would be to have it enabled, but I think it is not worth explicitly disabling it. It was necessary in the rbd-mirror case in order to decrease the memory usage, because rbd-mirror might have thousands of images opened simultaneously, but this is not the case here.